lib/appsignal/hooks/sidekiq.rb

```diff
@@ -14,23 +14,24 @@ class args retried_at failed_at
       end
 
       def call(_worker, item, _queue)
-        args =
-          if item["class"] == "ActiveJob::QueueAdapters::SidekiqAdapter::JobWrapper"
-            item["args"].first["arguments"]
-          else
-            item["args"]
-          end
-        params = Appsignal::Utils::ParamsSanitizer.sanitize args,
+        job = ::Sidekiq::Job.new(item)
+
+        # job.display_class needs to be called before job.display_args,
+        # see https://github.com/appsignal/appsignal-ruby/pull/348#issuecomment-333629065
+        display_class, display_method = job.display_class.split(/\.|#/, 2)
+        params = Appsignal::Utils::ParamsSanitizer.sanitize(
+          job.display_args,
           :filter_parameters => Appsignal.config[:filter_parameters]
+        )
 
         Appsignal.monitor_transaction(
           "perform_job.sidekiq",
-          :class       => item["wrapped"] || item["class"],
-          :method      => "perform",
+          :class       => display_class,
+          :method      => display_method || "perform",
```

Comment:
To show the difference between a class and instance method here for the Sidekiq delayed extension we might need to drop `Appsignal.monitor_transaction` usage and manually create a transaction.

The implementation in the [`Transaction#set_http_or_background_action` method](https://github.com/appsignal/appsignal-ruby/blob/eb421a537b8ebaac0d66fc20f002276e9998b489/lib/appsignal/transaction.rb#L225) always assumes a instance method.

Follow up from my reply here: https://github.com/mperham/sidekiq/pull/3623#pullrequestreview-66724350